### PR TITLE
ADAP-814: Add support for replacing materialized views with tables/views and vice versa

### PR DESCRIPTION
**Pull request by mikealfare**
### Problem

Materialized view support for dbt-postgres does not allow for replacing a materialized view with a table/view and vice versa.

### Solution

- Make the drop and rename statements generic across relations
- Dispatch to the relation-specific statement for drop/rename to allow for variations in the statement
- Allow a table/view to trigger the drop/rename of a materialized view
- Allow a materialized view to trigger the drop/rename of a table/view

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [x] I have run this code in development and it appears to resolve the stated issue  
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

File: core/dbt/include/global_project/macros/relations/replace.sql
Comment (mikealfare) [inline core/dbt/include/global_project/macros/relations/replace.sql:1]: Rename some of the macros to be more consistent (get_create_intermediate_sql, get_create_backup_sql, get_rename_intermediate_sql, get_drop_backup_sql).

File: core/dbt/adapters/base/relation.py [hunk 1]
```diff
@@ -35,6 +36,7 @@ class BaseRelation(FakeAPIObject, Hashable):
     include_policy: Policy = field(default_factory=lambda: Policy())
     quote_policy: Policy = field(default_factory=lambda: Policy())
     dbt_created: bool = False
+    relations_that_can_be_renamed: List[StrEnum] = field(default_factory=list)
```
Comment (mikealfare) [inline core/dbt/adapters/base/relation.py:39]: We should default to tables and views being rename-able.

Comment (mikealfare) [reply core/dbt/adapters/base/relation.py:39]: Rename to renameable relations.

Comment (mikealfare) [reply core/dbt/adapters/base/relation.py:39]: Add a comment regarding what this is.

**Review by colin-rogers-dbt — APPROVED** [review]

**Pull request merged**